test/server/services/contact_lists.js

```diff
@@ -0,0 +1,119 @@
+          table.timestamp('created_at').defaultTo(bookshelf.knex.fn.now());
+          table.timestamp('updated_at').defaultTo(bookshelf.knex.fn.now());
+        }).then(() => {
+          console.log(('** created contact_lists table'));
+          done();
+        }).catch((err) => {
+          console.log('! error creating contacts_lists table', err);
+        });
+      }
+    });
+  });
+  after(() =>
+    bookshelf.knex.schema.dropTable('contact_lists')
+      .then(() => {
+        console.log('* dropped contact_lists table');
+      })
+      .catch((err) => {
+        console.log('! error dropping contact_lists table', err);
+      })
+  );
+
+  describe('Data insertion', function() {
+    beforeEach(() => {
+      this.params = {
+        name: 'Millbrae Phone List'
+      };
+      this.params2 = {
+        name: 'Mission Bay Phone List'
+      };
+    });
+
+    it('should create contact list and save all parameters', (done) => {
+      ContactList.saveNewContactList(this.params, contactListsModel)
+        .then((contactList) => {
+          expect(contactList.attributes.name).to.equal(this.params.name);
+          done();
+        }).catch((err) => {
+          console.log('error with saving contact list test', err);
+        });
+    });
+  });
+  describe('Data update', function() {
+    beforeEach(() => {
+      this.params = {
+        name: 'San Francisco Mission Phone List'
+      };
+    });
+
+    it('should update a contact list name', (done) => {
+      ContactList.saveNewContactList(this.params, contactListsModel)
+        .then((contactList) => {
+          const newParams = { id: contactList.id, name: 'New Name' };
+          ContactList.updateContactListById(newParams, contactListsModel)
+            .then((updatedContactList) => {
+              expect(updatedContactList.attributes.name).to.equal(newParams.name);
+            })
+            .catch((err) => {
+              console.log('error with updating contact list \n\n');
+              done(err);
+            });
+          done();
+        })
+        .catch((err) => {
+            console.log('error with saving contact list before updating \n\n');
+            done(err);
+        });
+    });
+  });
+
+  describe('Data retrieval', function() {
+    beforeEach(() => {
+      this.params = {
+        id: 1
+      };
+    });
+    it('should retrieva a contact list by name', (done) => {
+      ContactList.getContactListById(this.params, contactListsModel)
+        .then((contactList) => {
+          expect(contactList.attributes.id).to.equal(1);
+          done();
+        })
+        .catch((err) => {
+            console.log('error with getting contact list \n\n');
+            done(err);
+        });
+    });
+    it('should retrieve all contact lists', (done) => {
+      ContactList.getAllContactLists(contactListsModel)
+        .then((contactLists) => {
+          expect(contactLists.length).to.equal(2);
```

Comment:
I dropped the table at the very end of the tests, so the lists from the previous `it` blocks are still present. Should I format this differently?